### PR TITLE
Feature detector extensions

### PR DESCRIPTION
**Pull request by pulkkins**
This pull request introduces some improvement to the `pysteps.feature` module:
- Add more documentation to `pysteps.feature.interface`.
- Implement the max_num_features keyword argument for all feature detectors. With this argument, the user can specify the maximum number of detected features. The features are chosen based on their importance, which depends on the used method.
- Add tests for the shitomasi and blob detectors.

### CODE REVIEW

File: pysteps/feature/tstorm.py [hunk 1]
```diff
@@ -235,6 +250,7 @@ def get_profile(areas, binary, ref, loc_max, time, minref):
         cells_id.cen_x.iloc[n] = int(np.nanmean(cells_id.x[n]))  # int(x[0])
         cells_id.cen_y.iloc[n] = int(np.nanmean(cells_id.y[n]))  # int(y[0])
         cells_id.max_ref.iloc[n] = maxref
+        cells_id.area.iloc[n] = len(cells_id.x.iloc[n])
```
Comment (dnerini) [inline pysteps/feature/tstorm.py:253]: is the area approximated by the number of x coordinates, or am I getting this code line wrong?

**Comment by dnerini** [comment]
very nice contribution, I have just one question regarding the way the tstorm features are ranked.

File: pysteps/feature/tstorm.py (hunk 1 above)
Comment (pulkkins) [reply pysteps/feature/tstorm.py:253]: Yes, the cell area is approximated by the number of pixels (i.e. the number of x- or y-coordinates) having the cell label. Is this OK?

Comment (dnerini) [reply pysteps/feature/tstorm.py:253]: Uh, right, very good, thanks!

**Review by dnerini — APPROVED** [review]

**Pull request merged**